collectington/collectington_api.py

```diff
@@ -168,12 +168,12 @@ def get_data_from_store(self, name_of_datastore):
 
     def _init_p_method(self, p_method, api_metric):
         """Internal method to metric methods with labels only if they're provided."""
-        if (
-            self.config.get("prometheus_metric_labels") is not None
-            and api_metric in self.config["prometheus_metric_labels"]
-        ):
-            labels = self.config["prometheus_metric_labels"][api_metric]
-            return p_method(api_metric, api_metric, labels)
+        for service in self.config["services"]:
+            if (self.config["services"][service].get("prometheus_metric_labels") is not None
+                and api_metric in self.config["services"][service]["prometheus_metric_labels"]
+            ):
+                labels = self.config["services"][service]["prometheus_metric_labels"][api_metric]
+                return p_method(api_metric, api_metric, labels)
```

Comment:
`self.config["services"][self.service_name]["prometheus_metric_labels"]` should work instead of for loop

Reply:
I see, ok - if I can get the service name without searching for it, all the better, thank you.